impl/src/planning.rs

```diff
@@ -1054,7 +1174,7 @@ mod checks {
 
   pub fn warn_unused_settings(
     all_crate_settings: &HashMap<String, CrateSettingsPerVersion>,
-    all_packages: &[Package],
+    all_packages: &Vec<&Package>,
```

Comment:
BTW, I've seen you make this `&[T]` --> `&Vec<T>` transition in a few places - clippy generally advises [against doing this](https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg), as it prohibits non-vec slices from being used when they might otherwise be valid as an argument.

Reply:
The change I was actually trying to make was `&[Package]` -> `&[&Package]`. I didn't know that would work (still learning about the finer details of Rust 😅 

I've updated this. Are there other places? I might be overlooking them.